core/ir/instr_shared.c

```diff
@@ -3733,21 +3744,40 @@ instr_is_reg_spill_or_restore_ex(void *drcontext, instr_t *instr, bool DR_only,
     return false;
 }
 
+DR_API
+bool
+instr_is_load_mcontext_base(instr_t *inst)
+{
+    if (instr_get_opcode(inst) != OP_load || !opnd_is_base_disp(instr_get_src(inst, 0)))
```

Comment:
This is not strict enough: for x86 it should check for a *far* base disp with the TLS segment; for arm, the stolen reg base.